openmc/deplete/abc.py

```diff
@@ -638,27 +650,59 @@ def __init__(self, operator, timesteps, power=None, power_density=None):
                         self._num_stages))
         self.operator = operator
         self.chain = operator.chain
-        if not isinstance(timesteps, Iterable):
-            self.timesteps = [timesteps]
-        else:
-            self.timesteps = timesteps
+
+        # Determine power and normalize units to W
         if power is None:
             if power_density is None:
                 raise ValueError("Either power or power density must be set")
             if not isinstance(power_density, Iterable):
                 power = power_density * operator.heavy_metal
             else:
-                power = [p * operator.heavy_metal for p in power_density]
-
+                power = [p*operator.heavy_metal for p in power_density]
         if not isinstance(power, Iterable):
             # Ensure that power is single value if that is the case
-            power = [power] * len(self.timesteps)
-        elif len(power) != len(self.timesteps):
+            power = [power] * len(timesteps)
+
+        if len(power) != len(timesteps):
             raise ValueError(
-                "Number of time steps != number of powers. {} vs {}".format(
-                    len(self.timesteps), len(power)))
+                "Number of time steps ({}) != number of powers ({})".format(
+                    len(timesteps), len(power)))
 
-        self.power = power
+        # Get list of times / units
+        if isinstance(timesteps[0], Iterable):
+            times, units = zip(*timesteps)
+        else:
+            times = timesteps
+            units = [timestep_units] * len(timesteps)
+
+        # Determine number of seconds for each timestep
+        seconds = []
+        for time, unit, watts in zip(times, units, power):
+            # Make sure values passed make sense
+            check_type('timestep', time, Real)
+            check_greater_than('timestep', time, 0.0, True)
+            check_type('timestep units', unit, str)
+            check_type('power', watts, Real)
+            check_greater_than('power', watts, 0.0, True)
+
+            if unit in ('s', 'sec'):
+                seconds.append(time)
+            elif unit in ('min', 'minute'):
+                seconds.append(time*_SECONDS_PER_MINUTE)
+            elif unit in ('h', 'hr', 'hour'):
+                seconds.append(time*_SECONDS_PER_HOUR)
+            elif unit in ('d', 'day'):
+                seconds.append(time*_SECONDS_PER_DAY)
+            elif unit.lower() == 'mwd/kg':
```

Comment:
Just a very small idea: can we add `GWd/t` as the alternative unit of burnup, though its conversion from `MWd/kg` is super easy.

Reply:
They are the same, no?

MWd/kg * (1 GWd)/(1000 MWd) * (1000 kg)/(1 tonne) = GWd/t

My personal opinion is that we should not include GWd/t as an option because "ton" is not well defined (ask someone how much a ton is and you may get different answers). MWd/kg on the other hand is unambiguous. I opted to not include "year" as an option for the same reason; there is no single answer to the question "how long is a year?" (365 d? 365.25 d? 365.2425 d?)

Reply:
They are the same. I come up with it because `GWd/t` is commonly used in many reference papers or design reports.  I agree with your idea on "year". As I know, "ton" in `GWd/tHM` or `MWd/tHM` is usually "metric ton". Anyway, this way can avoid such kind of "misunderstanding". So this is just my tiny question.  
Great job. Thanks.